framework/src/postprocessors/RelativeSolutionDifferenceNorm.C

```diff
@@ -31,5 +31,5 @@ RelativeSolutionDifferenceNorm::RelativeSolutionDifferenceNorm(const InputParame
 Real
 RelativeSolutionDifferenceNorm::getValue()
 {
-  return _fe_problem.relativeSolutionDifferenceNorm();
+  return dynamic_cast<Transient *>(_app.getExecutioner())->relativeSolutionDifferenceNorm();
```

Comment:
You are right. This needs to be done in constructor and errors out when the cast is not successful.